iOS-DOCS.md

```diff
@@ -0,0 +1,393 @@
+
+![choice card](media/choice_card.png)
+
+_**Fig. A** example true[X] mid-roll choice card_
+
+If the user opts for a normal ad break, or if the user does not make a selection before the countdown timer expires, the true[X] UI will close and playback of normal video ads can continue as usual.
+
+If the user opts to interact with true[X], an interactive ad unit will be shown to the user:
+
+![ad](media/ad.png)
+
+_**Fig. B** example true[X] interactive ad unit_
+
+The requirement for the user to "complete" this ad is for them to spend at least the allotted time on the unit and for at least one interaction (e.g. navigating anywhere through the ad).
+
+![true attention timer example](media/true-attention-timer-example.png)
+
+_**Fig. C** example true[X] attention timer_
+
+Once the user fulfills both requirements, a "I'm Done" button will appear in the bottom right, which the user can select to exit the true[X] ad. Having completed a true[X] ad, the user will be returned directly to content, skipping the remaining ads in the current ad pod.
+
+The Sponsored Stream flow is quite similar. In this scenario, a user will be shown a choice-card in the pre-roll:
+
+![sponsored card](media/sponsored_card.png)
+
+_**Fig. D** example true[X] pre-roll choice card (full-stream replacement)_
+
+Similarly, if the user opts-in and completes the true[X] ad, they will be skipped over the remainder of the pre-roll ad break. However, every subsequent mid-roll break in the current stream will also be skipped over. In this case instead of the regular pod of video ads, the user will be shown a "hero card" (also known as a "skip card"):
+
+![skip card](media/skip_card.png)
+
+_**Fig. E** example true[X] mid-roll skip card_
+
+This messaging will be displayed to the user for several seconds, after which they will be returned directly to content.
+
+
+## How to use TruexAdRenderer
+
+### When to show true[X]
+
+Upon receiving an ad schedule from your SSAI service, you should be able to detect whether or not true[X] is returned in any of the pods. true[X] ads should have `apiFramework` set to `VPAID` or `truex`.
+
+SSAI vendors differ in the way they convey information back about ad schedules to clients. Certain vendors such as Verizon / Uplynk expose API’s which return details about the ad schedule in a JSON object. For other vendors, for instance Google DAI, the true[X] payload will be encapsulated as part of a companion payload on the returned VAST ad. Please work with your true[X] point of contact if you have difficulty identifying the right approach to detecting the true[X] placeholder, which will be the trigger point for the ad experience.
+
+Once the player reaches a true[X] placeholder, it should pause, instantiate the `TruexAdRenderer` and immediately call [`init`](#initwithurl) followed by [`start`](#start).
```

Comment:
is the method called `init` or `initWithUrl`? 

We should use the actual code name

can't we just call it init after all?

Reply:
it is called `initWithUrl:adParameters:slotType:`
this is how Objective C style of `init(url, adParameters, slotType)`...